### ISSUE TITLE
Sorting the Sankey chart nodes.

### ISSUE THREAD

**Issue by VahidBo** [recharts/recharts#2164]
I want to sort my nodes in Sankey chart. How can I do that?
For example, for the following data:
```
const data0 = {
  nodes: [
    { name: "L0" },
    { name: "L1" },
    { name: "L2" },
    { name: "L3" },
    { name: "L4" },
    { name: "R5" },
    { name: "R6" },
    { name: "R7" },
    { name: "R8" },
    { name: "R9" }
  ],
  links: [
    { source: 0, target: 5, value: 30 },
    { source: 1, target: 8, value: 99 },
    { source: 1, target: 7, value: 20 },
    { source: 1, target: 6, value: 15 },
    { source: 4, target: 5, value: 6 },
    { source: 2, target: 8, value: 30 },
    { source: 0, target: 6, value: 15 },
    { source: 2, target: 9, value: 11 },
    { source: 3, target: 9, value: 8 },
    { source: 3, target: 8, value: 23 },
    { source: 2, target: 5, value: 20 }
  ]
};
```
For this data, the Sankey chart will be as follows:
![image](https://user-images.githubusercontent.com/15084663/84264927-49ec5f00-ab37-11ea-926f-debf48263f22.png)
But I want to sort the nods on each side. I want nodes to be on each side in this way:
![image](https://user-images.githubusercontent.com/15084663/84265550-43121c00-ab38-11ea-8ec9-9acc98c4ac86.png)
How can I do that?

**Comment by kunalshewale** [comment]
Hi, Can you please illustrate how you have achieved multicolour for the links in the above chart?

**Comment by VahidBo** [comment]
@kunalshewale You can pass the `link` prop to `Sankey` and the link could be `ReactElement`. You may define the background color of the link in that component. [This](https://github.com/recharts/recharts/blob/master/demo/component/DemoSankeyLink.js) is an example of a customized link component.

**Comment by zotijs** [comment]
@VahidBo did you managed to sort the nodes?

**Comment by dillonreedy** [comment]
@VahidBo I managed to achieve what you're looking to do in [this codesandbox](https://codesandbox.io/s/gifted-fog-1dutf).
![image](https://user-images.githubusercontent.com/3454995/103985728-16b29280-514f-11eb-80e9-20eb9269cb45.png)


@kunalshewale I achieved the node order through setting the iterations prop on the Sankey to zero: `<Sankey iterations={0}>`. This causes the sorting done by the Sankey component not to happen. Thus sort your data however you want before you connect it as a prop to the Sankey component.

@zotijs As far as the node colors I pass a custom colors prop to my node component `<DemoSankeyNode  colors={colors} />`. Inside the custom node component it just fills the rectangle with whatever the nodes' index is modded by the total number of colors. You'll probably wanna make that a little more dynamic depending on the number of nodes you have on either side, and the number of colors you want.
